docs/guides/intro-to-tests.md

```diff
@@ -0,0 +1,154 @@
+
+```js
+test("will count the number of words for a single word string", function () {
+  const input = "hello";
+  const actualResult = createWordCount(input);
+  const expectedResult = {
+    hello: 1,
+  };
+  expect(actualResult).toEqual(expectedResult);
+});
+```
+
+Let's break down the different parts of this test code.
+
+## Anatomy of a test
+
+#### 1. Test description
+
+```js
+test("will count the number of words in a single word string", function () {});
+```
+
+We use a function in Jest called `test` to write a test description - `"will count the number of words for a single word string"` which describes the behaviour we expect in this particular test. There is also a function `function () {}` in which we will write our test code.
+
+#### 2. Test setup
+
+```js
+const input = "hello";
+const actualResult = createWordCount(input);
+const expectedResult = {
+  hello: 1,
+};
+```
+
+In this section, we're declaring some important values that we will need to compare later on in the test.
+We declare an `input` variable that stores a value that we will pass to the function we are testing - `createWordCount`.
+We declare an `actualResult` variable that stores the **return value** of our function `createWordCount`.
+We declare an `expectedResult` which stores what we **expect** the `actualResult` to be when our function is behaving according to the test description.
+
+❗ Note: The test setup will vary from test to test - so you can expect to see different variable names altogether in different tests. The setup here is done in order to make it clearer what values will be compared later on in the test.
+
+#### 3. Assertion
+
+```js
+expect(actualResult).toEqual(expectedResult);
+```
+
+The assertion is the part of the test code that actually checks to see if something is true. In this example, we are claiming that the following is true:
+
+> "We expect `actualResult` **to equal** `expectedResult`"
+
+Notice that the statement above is very similar to the syntax used in the test code.
+
+The function [`toEqual`](https://jestjs.io/docs/expect#toequalvalue) ( which is part of the Jest framework ) is used to check that `actualResult` ( `createWordCount`'s return value ) and `expectedResult` are equal to each other. There are [many other functions](https://jestjs.io/docs/using-matchers) like `toEqual` which we can use to make different assertions.
+
+### Pass ✅ or Fail ❌
+
+Finally, our test can either pass or fail.
+
+- If `actualResult` is equal to `expectedResult` then the test has passed ✅.
+- If `actualResult` is _not_ equal to `expectedResult` then the test has failed ❌.
+
```

Comment:
I think we should add something here along the lines of:

> **Don't panic if a test is failing!**
> Remember: tests are here to help you solve the problem! If you carefully read the test feedback it will tell you what the problem is.

Reply:
Excellent idea - love it :muscle: